modules/auth-on-llama-stack.adoc

```diff
@@ -0,0 +1,388 @@
+
+* Realm: `llamastack-demo`
+* Client: `llamastack` with direct access grants enabled
+* Role: `inference_max` grants access to restricted models and a protocol mapper that adds realm roles to the access token under the claim name `llamastack_roles`
+* Two test users:
+** `user1` as a basic user with no assigned roles
+** `user2` as an advanced user assigned the `inference_max` role
+* The client secret generated by Keycloak must be saved as you will need it for token requests.
+
+This document assumes the Keycloak server is available at `https://my-keycloak-server.com`
+
+
+.Prerequisites
+
+* You have installed {openshift-platform} 4.19 or newer.
+* You have logged in to {productname-long}.
+* You have cluster administrator privileges for your OpenShift cluster.
+* You have installed the {openshift-cli} as described in the appropriate documentation for your cluster:
+ifdef::upstream,self-managed[]
+** link:https://docs.redhat.com/en/documentation/openshift_container_platform/{ocp-latest-version}/html/cli_tools/openshift-cli-oc#installing-openshift-cli[Installing the OpenShift CLI^] for OpenShift Container Platform
+** link:https://docs.redhat.com/en/documentation/red_hat_openshift_service_on_aws/{rosa-latest-version}/html/cli_tools/openshift-cli-oc#installing-openshift-cli[Installing the OpenShift CLI^] for {rosa-productname}
+endif::[]
+ifdef::cloud-service[]
+** link:https://docs.redhat.com/en/documentation/openshift_dedicated/{osd-latest-version}/html/cli_tools/openshift-cli-oc#installing-openshift-cli[Installing the OpenShift CLI^] for OpenShift Dedicated
+** link:https://docs.redhat.com/en/documentation/red_hat_openshift_service_on_aws_classic_architecture/{rosa-classic-latest-version}/html/cli_tools/openshift-cli-oc#installing-openshift-cli[Installing the OpenShift CLI^] for {rosa-classic-productname}
+endif::[]
+
+.Procedure
+
+. To configure Llama Stack to use Role-Based Access Control (RBAC) to access models, you need to view and verify the OAuth provider token structure.
+
+.. Generate a KeyCloak test token to view the structure with the following command:
++
+[source,terminal]
+----
+$ curl -d client_id=llamastack -d client_secret=YOUR_CLIENT_SECRET -d username=user1 -d password=user-password -d grant_type=password https://my-keycloak-server.com/realms/llamastack-demo/protocol/openid-connect/token | jq -r .access_token > test.token
+----
+
+.. View the token claims by running the following command:
++
+[source,terminal]
+----
+$ cat test.token | cut -d . -f 2 | base64 -d 2>/dev/null | jq .
+----
++
+.Example token structure from KeyCloak
+[source,terminal]
+----
+$ {
+  "iss": "http://my-keycloak-server.com/realms/llamastack-demo",
+  "aud": "account",
+  "sub": "761cdc99-80e5-4506-9b9e-26a67a8566f7",
+  "preferred_username": "user1",
+  "llamastack_roles": [
+    "inference_max",
+  ],
+}
+----
+
+. You need to then create a `run.yaml` file that defines the necessary configurations for OAuth.
```

Comment:
Might consider rewording for conciseness: Create a `run.yaml` file that defines the necessary configurations for OAuth.